tools/nmr_bucketing/NmrBucketing_wrapper.R

```diff
@@ -0,0 +1,307 @@
+                sampleMetadataOut = paste(directory,"_NmrBucketing_sampleMetadata.tsv",sep=""),
+                variableMetadataOut = paste(directory,"_NmrBucketing_variableMetadata.tsv",sep=""),
+                graphOut = paste(directory,"_NmrBucketing_graph.pdf",sep=""),
+                logOut = paste(directory,"_NmrBucketing_log.txt",sep="")))
+}
+
+##------------------------------
+## Options
+##------------------------------
+strAsFacL <- options()$stringsAsFactors
+options(stringsAsFactors = FALSE)
+
+
+##------------------------------
+## Libraries laoding
+##------------------------------
+# For parseCommandArgs function
+library(batch)
+# For cumtrapz function
+library(pracma)
+
+# R script call
+source_local <- function(fname)
+{
+	argv <- commandArgs(trailingOnly = FALSE)
+	base_dir <- dirname(substring(argv[grep("--file=", argv)], 8))
+	source(paste(base_dir, fname, sep="/"))
+}
+#Import the different functions
+source_local("NmrBucketing_script.R")
+source_local("DrawSpec.R")
+
+##------------------------------
+## Errors ?????????????????????
+##------------------------------
+
+
+##------------------------------
+## Constants
+##------------------------------
+topEnvC <- environment()
+flagC <- "\n"
+
+
+##------------------------------
+## Script
+##------------------------------
+if(!runExampleL)
+    argLs <- parseCommandArgs(evaluate=FALSE)
+
+
+## Parameters Loading
+##-------------------
+  # Inputs
+if (!is.null(argLs[["zipfile"]])){
+	fileType="zip"
+	zipfile= argLs[["zipfile"]]
+	directory=unzip(zipfile, list=F)
+	directory=paste(getwd(),strsplit(directory[1],"/")[[1]][2],sep="/")
+} else if (!is.null(argLs[["tsvfile"]])){
+	fileType="tsv"
+	directory <- read.table(argLs[["tsvfile"]],check.names=FALSE,header=TRUE,sep="\t")
+}
+
+leftBorder <- argLs[["left_border"]]
+rightBorder <- argLs[["right_border"]]
+bucketSize <- argLs[["bucket_width"]]
+exclusionZones <- argLs[["zone_exclusion_choices.choice"]]
+
+exclusionZonesBorders <- NULL
+if (!is.null(argLs$zone_exclusion_left))
+{
+   for(i in which(names(argLs)=="zone_exclusion_left"))
+   {
+     exclusionZonesBorders <- c(exclusionZonesBorders,list(c(argLs[[i]],argLs[[i+1]])))
+   }
+}
+
+graphique <- argLs[["graphType"]]
+
+  # Outputs
+nomGraphe <- argLs[["graphOut"]]
+dataMatrixOut <- argLs[["dataMatrixOut"]]
+logFile <- argLs[["logOut"]]
+if (fileType=="zip")
+{
+  sampleMetadataOut <- argLs[["sampleOut"]]
+  variableMetadataOut <- argLs[["variableOut"]]
+}
+
+
+## Checking R packages
+##--------------------
+sink(logFile)
+cat("\t PACKAGE INFO \n")
+pkgs=c("batch","pracma")
+for(pkg in pkgs) {
+    suppressPackageStartupMessages( stopifnot( library(pkg, quietly=TRUE, logical.return=TRUE, character.only=TRUE)))
+    cat(pkg,"\t",as.character(packageVersion(pkg)),"\n",sep="")
+}
+cat("\n")
+
+
+## Checking arguments
+##-------------------
+error.stock <- "\n"
+
+if(length(error.stock) > 1)
+  stop(error.stock)
+

```

Comment:
this whole section could be removed without the command line parsing section and if following the approach recommended above.